src/index.js

```diff
@@ -154,20 +135,43 @@ export default function normalize(obj, key=null) {
     return { entities, normalizations, result };
 }
 
-export function denormalize(key, builder, state, requiredFields) {
+/**
+* Denormalizes the entity that is associated with the key.
+*
+* @param {Object} key
+* @param {Builder} builder
+* @param {Map} state
+* @param {Function} validator Accepts three parameters (denormalized entity, entity key, key) and returns false if entity is invalid
+* @return {Array}|{Object}|{Void} denormalized entity or null if invalid or not found
+*/
+export function denormalize(key, builder, state, validator = null) {
     const entityKey = getEntityKey(builder);
     if (!state.entities[entityKey]) {
         return;
     }
     if (Array.isArray(key)) {
-        return key.map(id => {
+        let valid = true;
+        let entities = [];
+        key.every(id => {
```

Comment:
oh awesome, i didn't read further into that description

On Tue, Jan 26, 2016 at 12:05 PM Felix Mo <notifications@github.com> wrote:

> In src/index.js
> <https://github.com/lunohq/protobuf-normalizr/pull/1#discussion_r50891982>
> :
>
> >      const entityKey = getEntityKey(builder);
> >      if (!state.entities[entityKey]) {
> >          return;
> >      }
> >      if (Array.isArray(key)) {
> > -        return key.map(id => {
> > +        let valid = true;
> > +        let entities = [];
> > +        key.every(id => {
>
> I went back and simplified the logic a bit in 2d24e8e
> <https://github.com/lunohq/protobuf-normalizr/commit/2d24e8ef97ec5e38e83d0d5afe57895679731241>.
>
>
> But I think every should bail after the first invalid entity.
>
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lunohq/protobuf-normalizr/pull/1/files#r50891982>.
>